gammapy/datasets/flux_points.py

```diff
@@ -17,6 +17,20 @@
 __all__ = ["FluxPointsDataset"]
 
 
+def _get_reference_model(model, energy_bounds, margin_percent=70):
+    if isinstance(model.spatial_model, TemplateSpatialModel):
```

Comment:
Technically, this spatial integration should be performed for any spatial model that is energy dependent right? Maybe this should be the criterion applied?

Reply:
Or maybe also if the model does not integrate to unity...

Reply:
in the case the geom could re-use what we have for spatial model plotting :https://github.com/gammapy/gammapy/blob/0afcf2ab0bbfae1559b0546c156e9544e7c19389/gammapy/modeling/models/spatial.py#L277-L284

Reply:
Let's keep that for a separate PR.